block-node/messaging/src/main/java/org/hiero/block/server/messaging/impl/MessagingServiceImpl.java

```diff
@@ -0,0 +1,300 @@
+        // keep track of the event processor & thread so we can stop them later
+        dynamicNoBackPressureBlockItemHandlerToEventProcessor.put(handler, batchEventProcessor);
+        dynamicNoBackPressureBlockItemHandlers.put(handler, handlerThread);
+        // start the event processor thread
+        handlerThread.start();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void unregisterDynamicNoBackpressureBlockItemHandler(NoBackPressureBlockItemHandler handler) {
+        final Thread handlerThread = dynamicNoBackPressureBlockItemHandlers.remove(handler);
+        final BatchEventProcessor<BlockItemBatchRingEvent> eventProcessor =
+                dynamicNoBackPressureBlockItemHandlerToEventProcessor.remove(handler);
+        if (eventProcessor != null) {
+            blockItemDisruptor.getRingBuffer().removeGatingSequence(eventProcessor.getSequence());
+            // stop the event processor
+            eventProcessor.halt();
+        }
+        // interrupt the thread so it stops quickly
+        if (handlerThread != null) {
+            handlerThread.interrupt();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void sendBlockNotification(BlockNotification notification) {
+        blockNotificationDisruptor.getRingBuffer().publishEvent((event, sequence) -> event.set(notification));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void registerBlockNotificationHandler(BlockNotificationHandler handler) throws IllegalStateException {
+        blockNotificationHandlers.add((event, sequence, endOfBatch) -> handler.handleBlockNotification(event.get()));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() {
+        // Register the block item handlers with the disruptor
+        //noinspection unchecked
+        blockItemDisruptor.handleEventsWith(blockItemHandlers.toArray(new EventHandler[0]));
+        // Register the block notification handlers with the disruptor
+        //noinspection unchecked
+        blockNotificationDisruptor.handleEventsWith(blockNotificationHandlers.toArray(new EventHandler[0]));
+        // Set the exception handler for the disruptors
+        blockItemDisruptor.setDefaultExceptionHandler(BLOCK_ITEM_EXCEPTION_HANDLER);
+        blockNotificationDisruptor.setDefaultExceptionHandler(BLOCK_NOTIFICATION_EXCEPTION_HANDLER);
+        // start the disruptors
+        blockItemDisruptor.start();
+        blockNotificationDisruptor.start();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void shutdown() {
+        // Shuts down all the threads handling events.
+        blockItemDisruptor.shutdown();
+        blockNotificationDisruptor.shutdown();
+        // Stop all the dynamic no back pressure block item handlers
+        for (Thread thread : dynamicNoBackPressureBlockItemHandlers.values()) {
+            thread.interrupt();
+        }
```

Comment:
Q: Should there be logic here to unregister the non dynamic bloc handlers - per the comments on `shutdown()` in the interface.

Reply:
explained above